Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -119,62 +122,209 @@ def get_current_utc_time() -> datetime:
 
+    if user:
+        demisto.debug(f"User {user_to_search} was found.")
+        return format_user_results(user)
+    else:
+        demisto.debug(format_user_not_found_error(user=user_to_search))
+        return {}
+
+
+def format_user_results(user: dict):
+    """
+    Formats and truncates the user result from the Slack API
+
+    Args:
+        user: The user response retrieved from the Slack API
+    Returns:
+        Formatted and truncated version of the result which is context safe.
+    """
+    return {
+        'name': user.get('name'),
+        'id': user.get('id'),
+        'profile': {
+            'email': user.get('profile', {}).get('email', ''),
+            'real_name': user.get('profile', {}).get('real_name', ''),
+            'display_name': user.get('profile', {}).get('display_name', ''),
+        }
+    }
+
+
+def get_user_by_name_caching_disabled(user_to_search: str) -> dict:
+    """
+    When Disable Caching is true, we look for the user by email only.
+
+    Args:
+        user_to_search: The user's email
+    Returns:
+        A slack user object
+    """
+    user = {}
+    if re.match(emailRegex, user_to_search):
+        demisto.debug(f"Checking via API for email of {user_to_search}")
+        user = get_user_by_email(user_to_search)
+    return user
+
+
+def get_user_by_name_caching_enabled(user_to_search, add_to_context):
+    """
+    When Disable Caching is false, we look for the user first in the context. If the user is not found, then we proceed to
+    search for the user by email, if that fails, then we will search for the user using a paginated call.
+
+    Args:
+        user_to_search: The user's email
+    Returns:
+        A slack user object
+    """
+    user = {}
+    integration_context = get_integration_context(SYNC_CONTEXT)
+    if integration_context.get('users'):
+        demisto.debug(f"Checking in context for {user_to_search}")
+        # Check if we already have the user to prevent call to users.lookupByEmail
+        users = json.loads(integration_context['users'])
+        user = return_user_filter(user_to_search.lower(), users)
+
+    if not user and re.match(emailRegex, user_to_search):
+        demisto.debug(f"Checking via API for email of {user_to_search}")
+        user = get_user_by_email(user_to_search)
+        if user and (add_to_context or not DISABLE_CACHING):
+            integration_context = get_integration_context(SYNC_CONTEXT)
+            add_user_to_context(user=user, integration_context=integration_context)
+    if not user:
+        demisto.debug(f"Couldn't find {user_to_search} and caching is disabled. Checking API")
+        user = paginated_search_for_user(user_to_search)
+        demisto.debug(f"Found {user_to_search} - {user}")
+        if user and (add_to_context or not DISABLE_CACHING):
+            integration_context = get_integration_context(SYNC_CONTEXT)
+            add_user_to_context(user=user, integration_context=integration_context)
```

Comment:
please review this code.
the name fo the function is `get_user_by_name_caching_enabled` 
then you should not be checking DISABLE_CACHING inside.